### PR TITLE
issue #4449 Fix the issue using Arn field instead of Name in IamInstanceProfileSpecification

### PR DESCRIPTION
**Pull request by saravanan30erd**
Used Arn field instead of Name in Type IamInstanceProfileSpecification.

### CODE REVIEW

**Comment by bflad** [comment]
Changing the parameter it uses will likely break existing configurations. 😄  Can you please introduce a separate Terraform attribute into the schema? e.g. `iam_instance_profile_arn` Thanks!

File: aws/resource_aws_spot_fleet_request.go [hunk 1]
```diff
@@ -191,6 +191,11 @@ func resourceAwsSpotFleetRequest() *schema.Resource {
 							ForceNew: true,
 							Optional: true,
 						},
+						"iam_instance_profile_arn": {
```
Comment (bflad) [inline aws/resource_aws_spot_fleet_request.go]: Can you please
* Document the new argument in the argument reference near the existing `iam_instance_profile` argument
* Add a simple acceptance test to cover this new argument? Copy pasting an existing test and test configuration that uses `iam_instance_profile` is fine 👍

File: website/docs/r/spot_fleet_request.html.markdown [hunk 2]
```diff
@@ -27,13 +27,15 @@ resource "aws_spot_fleet_request" "cheap_compute" {
     ami               = "ami-1234"
     spot_price        = "2.793"
     placement_tenancy = "dedicated"
+    iam_instance_profile_arn = "arn:aws:iam::12345678:instance-profile/webserver-role"
```
Comment (bflad) [inline website/docs/r/spot_fleet_request.html.markdown]: We should either fix this so its a properly formatted ARN (account ID is 12 digits `123456789012`) or just reference a "fake" resource (`"${aws_iam_instance_profile.example.arn}"`). Also as a nitpick, if these are going to be added in here, please line up all the equals symbols similar to how `terraform fmt` would e.g.

```
ami                      = "ami-1234"
spot_price               = "2.793"
placement_tenancy        = "dedicated"
iam_instance_profile_arn = "arn:aws:iam::12345678:instance-profile/webserver-role"
```

File: aws/resource_aws_spot_fleet_request.go (hunk 1 above)
Comment (saravanan30erd) [reply aws/resource_aws_spot_fleet_request.go]: Documented the new argument under `launch_specification` section and added the acceptance test.

File: website/docs/r/spot_fleet_request.html.markdown (hunk 2 above)
Comment (saravanan30erd) [reply website/docs/r/spot_fleet_request.html.markdown]: Done 👍

**Comment by bflad** [comment]
This has been released in [version 1.19.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1190-may-16-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by jekh** [comment]
I recently upgraded to 1.19.0 without changing my existing `iam_instance_profile`s to `iam_instance_profile_arn`s. As a result, none of the launched instances had IAM roles assigned at all. So this definitely will break existing configurations.

Changing to the new _arn attribute fixed the problem, and using the ARN seems like a fine idea. Would it be better to simply remove the existing `iam_instance_profile` attribute, so at least folks' terraform breaks, rather than silently not working properly in a non-obvious way?

**Comment by saravanan30erd** [comment]
when using `iam_instance_profile`, `iam_instance_profile_arn` assigned to empty string and looks like it overrides the `IamInstanceProfile ` finally so that launched instances had no IAM roles assigned at all.

**Comment by bflad** [comment]
This has been released in [version 1.20.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1200-may-23-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**